integration_tests/e2e/goal/reviewUpdateGoal.cy.ts

```diff
@@ -0,0 +1,103 @@
+import Page from '../../pages/page'
+import UpdateGoalPage from '../../pages/goal/UpdateGoalPage'
+import ReviewUpdateGoalPage from '../../pages/goal/ReviewUpdateGoalPage'
+import OverviewPage from '../../pages/overview/OverviewPage'
+import AuthorisationErrorPage from '../../pages/authorisationError'
+
+context('Review updated goal', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignInAsUserWithEditAuthority')
+    cy.task('stubAuthUser')
+    cy.task('getPrisonerById', 'G6115VJ')
+    cy.task('getActionPlan', 'G6115VJ')
+    cy.task('stubLearnerProfile')
+    cy.task('updateGoal')
+  })
+
+  it('should not be able to navigate directly to Review Goal given previous forms have not been submitted', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+    cy.signIn()
+
+    // When
+    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/update/review`)
+
+    // Then
+    const overviewPage = Page.verifyOnPage(OverviewPage)
+    overviewPage.isForPrisoner(prisonNumber)
+  })
+
+  it('should navigate back to the update goals form if the go back to edit goal button is clicked', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+    cy.signIn()
+
+    // When
```

Comment:
I think the scenarios you are testing in these cypress tests are good 👍 ; its just that the Given, When, Then's are very slightly confused.

The `When` is the thing you do in order to cause the behaviour and ideally should be a single function call (though in some cases thats not always possible)

In these tests, the key action is clicking the back button on the Update Goal Review page Everything before the actual click of the button (ie. how we got to the page) is all setup/pre-amble, and so lives in the `Given` block

Everything after the submit button has been clicked are your assertions so all live under a single `Then` comment block.

Like this:
```
    // Given
    const prisonNumber = 'G6115VJ'
    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
    cy.signIn()

    cy.visit(`/plan/${prisonNumber}/view/overview`)
    const overviewPage = Page.verifyOnPage(OverviewPage)
    const updateGoalPage = overviewPage.clickUpdateButtonForFirstGoal()
    updateGoalPage.isForGoal(goalReference).submitPage()

    const reviewUpdateGoalPage = Page.verifyOnPage(ReviewUpdateGoalPage)
    reviewUpdateGoalPage.isForPrisoner(prisonNumber).goBackToEditGoal()

    // When
    reviewUpdateGoalPage.goBackToEditGoal()

    // Then
    Page.verifyOnPage(UpdateGoalPage).isForGoal(goalReference)
```

If you can apply that approach to the other scenarios in this test that would be great 👍 
Shout if you need help

Reply:
Removed the duplicated `// Then` blocks